internal/provider/data_source_archive_file_test.go

```diff
@@ -360,6 +360,604 @@ func TestAccArchiveFile_SourceConfigConflicting(t *testing.T) {
 	})
 }
 
+func TestAccArchiveFile_IncludeSymlinkDirectories(t *testing.T) {
+	td := t.TempDir()
+
+	f := filepath.Join(td, "zip_file_acc_test.zip")
+
+	symlinkFile := "test-fixtures/test-dir-with-symlink-file/test-symlink.txt"
+	symlinkFileAbs, err := filepath.Abs(symlinkFile)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	symlinkDirWithRegularFiles := "test-fixtures/test-symlink-dir"
+	symlinkDirWithRegularFilesAbs, err := filepath.Abs(symlinkDirWithRegularFiles)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	symlinkDirWithSymlinkFiles := "test-fixtures/test-dir-with-symlink-file"
+	symlinkDirWithSymlinkFilesAbs, err := filepath.Abs(symlinkDirWithSymlinkFiles)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	symlinkFileInSymlinkDir := "test-fixtures/test-symlink-dir-with-symlink-file/test-symlink.txt"
+	symlinkFileInSymlinkDirAbs, err := filepath.Abs(symlinkFileInSymlinkDir)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	symlinkDirInRegularDir := "test-fixtures/test-dir-with-symlink-dir"
+	symlinkDirInRegularDirAbs, err := filepath.Abs(symlinkDirInRegularDir)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	multipleDirsAndFiles := "test-fixtures"
+	multipleDirsAndFilesAbs, err := filepath.Abs(multipleDirsAndFiles)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var fileSize string
+	r.ParallelTest(t, r.TestCase{
+		ProtoV5ProviderFactories: protoV5ProviderFactories(),
+		Steps: []r.TestStep{
```

Comment:
Would this testing be a little easier to read/grok/parallelize if each test step was treated as its own unit test? This would help ensure there is a separate `t.TempDir()` for each archive and there could be code comments explaining what should (or shouldn't) be expected for each test.

Reply:
Sounds like a good idea. Have separated each of the test steps into an individual test and added comments to clarify the expectations.